assets/react/controllers/Mosaic.jsx

```diff
@@ -1,66 +1,159 @@
-
-          <GridItem
-              variant={Data[randomExpose].variant}
-              description={Data[randomExpose].description}
-              image={Data[randomExpose].image}
-              exposed
-              tileIcons={Settings.exposeIcon}
-              tileBorders={Settings.exposeBorder}
-          />
-
-          {Settings.mosaicLogo && <Logo />}
-
-          <GlobalStyles />
-      </div>
-  );
-}
+import LoaderSrc from "../assets/loader.svg";
+import {CenteredContent} from "../components/CenteredContent";
+
+function Mosaic() {
+    const TILES_LOADING_INTERVAL = 60 * 1000 * 5;
+
+    const [params, setParams] = useState(null);
+    const [screen, setScreen] = useState(null);
+    const [config, setConfig] = useState(null);
+    const [tiles, setTiles] = useState([]);
+    const [randomExpose, setRandomExpose] = useState(1);
+    const [errorMessage, setErrorMessage] = useState(null);
+
+    const loadScreen = () => {
+        fetch(`/api/v1/screens/${params.id}`, {
+            headers: {
+                authorization: `Bearer ${params.key}`
+            }
+        })
+            .then((resp) => {
+                if (!resp.ok) {
+                    throw new Error("Could not fetch screen config");
+                }
+
+                return resp.json();
+            })
+            .then((data) => setScreen(data))
+            .catch((err) => {
+                setErrorMessage(err.message);
+            });
+    }
+
+    const loadTiles = (numberOfTiles) => {
+        fetch(`/api/v1/tiles/random?page=1&limit=${numberOfTiles}`, {
+            headers: {
+                authorization: `Bearer ${params.key}`
+            }
+        })
+            .then((resp) => {
+                if (!resp.ok) {
+                    throw new Error("Could not fetch tiles");
+                }
+
+                return resp.json();
+            })
+            .then((data) => {
+                const loadedTiles = [...data['hydra:member']];
+                setTiles(loadedTiles.map((tile) => {
+                    tile.extra = JSON.parse(tile.extra);
+                    return tile;
+                }));
+            })
+            .catch((err) => {
+                setErrorMessage(err.message);
+            });
+    }
+
+    useEffect(() => {
+        const url = new URL(window.location.href);
+        const id = url.searchParams.get('id');
+        const key = url.searchParams.get('key');
+
+        if (id === null || key === null) {
+            setErrorMessage("Id and/or key are not set.")
+        } else {
+            setParams({id, key});
+        }
+    }, []);
+
+    useEffect(() => {
+        if (params === null) {
+            return;
+        }
+
+        loadScreen();
+    }, [params]);
+
+    useEffect(() => {
+        if (screen === null) {
+            return;
+        }
+
+        const numberOfTiles = screen.gridColumns * screen.gridRows;
+
+        setConfig({
+            gridColumns: screen.gridColumns ?? 6,
```

Comment:
Do we need fallback for gridColumns, both here and in GlobalStyles.js?

Reply:
Removed in global styles when we know they are initialized in config.